graphistry/compute/python_remote.py

```diff
@@ -0,0 +1,103 @@
+from typing import Optional, Union
+from typing_extensions import Literal
+import ast
+import pandas as pd
+import requests
+
+from graphistry.Plottable import Plottable
+
+
+def python_remote(
+    self: Plottable,
+    code: str,
+    api_token: Optional[str] = None,
+    dataset_id: Optional[str] = None,
+    engine: Literal["pandas", "cudf"] = "cudf",
+    validate: bool = True
+) -> Union[Plottable, pd.DataFrame]:
+    """Remotely run Python code on a remote dataset.
+    
+    Uses the latest bound `_dataset_id`, and uploads current dataset if not already bound. Note that rebinding calls of `edges()` and `nodes()` reset the `_dataset_id` binding.
+
+    :param code: Python code that includes a top-level function `def task(g: Plottable) -> Union[str, Dict]`.
+    :type code: str
+
+    :param api_token: Optional JWT token. If not provided, refreshes JWT and uses that.
+    :type api_token: Optional[str]
+
+    :param dataset_id: Optional dataset_id. If not provided, will uplaod current data, store that dataset_id, and run GFQL against that.
+    :type dataset_id: Optional[str]
+
+    :param engine: Override which run mode GFQL uses. Defaults to "cudf".
+    :type engine: Literal["pandas", "cudf]
+
+    :param validate: Whether to locally test code, and if uploading data, the data. Default true.
+    :type validate: bool
+
+    **Example: Upload data and count the results**
+        ::
+
+            import graphistry
+            from graphistry import n, e
+            es = pandas.DataFrame({'src': [0,1,2], 'dst': [1,2,0]})
+            g1 = graphistry
+                .edges(es, source='src', destination='dst')
+                .upload()
+            assert g1._dataset_id is not None, "Successfully uploaded"
+            out_json = g1.python_remote(
+                code='''
+from typing import Any, Dict
```

Comment:
it is not necessary to start from 0 indents here, the python endpoint should normalize tabs automatically even if the remote code snipped starts from n > 1 indents.

just a nice thing for readability of snippet in context of other code